tail/tail.go

```diff
@@ -57,6 +58,46 @@ type State struct {
 	Offset int64
 }
 
+// GetSampledEntries wraps GetEntries and returns a list of channels that
+// provide sampled entries
+func GetSampledEntries(conf Config, sampleRate uint) ([]chan string, error) {
+	unsampledLinesChans, err := GetEntries(conf)
+	if err != nil {
+		return nil, err
+	}
+	if sampleRate == 1 {
+		return unsampledLinesChans, nil
+	}
+
+	sampledLinesChans := make([]chan string, 0, len(unsampledLinesChans))
+
+	for _, lines := range unsampledLinesChans {
+		sampledLines := make(chan string)
+		go func(pLines chan string) {
+			defer close(sampledLines)
+			for line := range pLines {
+				if shouldDrop(sampleRate) {
+					logrus.WithFields(logrus.Fields{
+						"line": line,
```

Comment:
if you're going to log, stick the `sampleRate` in there.

the debug output is getting a little unwieldy, but that's a point for another time, i think.